### PR TITLE
Closes #106 - Flag for tag in project bump

### LINKED ISSUE #106: Flag to create a tag on `project bump`
Command `dephell project bump` creates git tag by default. I think, it can be annoying for some users. Let's add `--tag` option to enable this feature. Without `--tag` (or `tag = true` in config, of course) it won't create tag.

See #104 for example how to do it.

### PR DESCRIPTION
**Pull request by nandahkrishna**
Refer issue #106 and please suggest changes.

### CODE REVIEW

File: dephell/commands/project_bump.py [hunk 1]
```diff
@@ -99,25 +99,26 @@ def __call__(self) -> bool:
...
-                self.logger.error('cannot commit files')
-                return False
-            ok = git_tag(
-                name='v.' + str(new_version),
-                project=project,
-            )
-            if not ok:
-                self.logger.error('cannot add tag into git repo')
-                return False
+        if self.config.get['tag']:
```
Comment (orsinium) [inline dephell/commands/project_bump.py]: `.get('tag')`

File: dephell/config/builders.py [hunk 2]
```diff
@@ -72,6 +72,7 @@ def build_other(parser):
     other_group = parser.add_argument_group('Other')
 
     other_group.add_argument('--owner', help='name of the owner.')
+    other_group.add_argument('--tag', help='create git tag.')
```
Comment (orsinium) [inline dephell/config/builders.py]: `action='store_true'`

And you forgot to add it into config's validation scheme

File: dephell/commands/project_bump.py (hunk 1 above)
Comment (nandahkrishna) [reply dephell/commands/project_bump.py]: Oops.

File: dephell/config/builders.py (hunk 2 above)
Comment (nandahkrishna) [reply dephell/config/builders.py]: Ah sure I'll do that.

**Comment by nandahkrishna** [comment]
I'll write the test and update this soon.

**Comment by nandahkrishna** [comment]
So far, in the test, I've called the command and passed the config. How do you check if the version has changed? Do you first read the version from the `from` or `project` file given in the config, then check if it's changed after the command is called?

**Comment by orsinium** [comment]
Beautiful start, I like it. Yeah, make `project/__init__.py` inside this dir, write version in it and check that it was changed after call :)

**Comment by nandahkrishna** [comment]
Okay, I created `project/__init__.py` and used `get_version_from_project` to get version values for the test. Not sure if I've done it right, do check and review!

**Comment by orsinium** [comment]
You haven't written a version in temp_path. This is the empty dir :)

```
(temp_path / 'project').mkdir()
init_path = (temp_path / 'project' / '__init__')
init_path.write_text('__version__ = "1.2.3"')
...
command = ProjectBumpCommand(argv=['fix'], config=config)
...
assert init_path.read_text() == '__version__ = "1.2.4"'
```

So, you don't need `dephell_discover` here.

**Comment by orsinium** [comment]
Do you understand how this test works? Instead of the current project, you make an empty dir and point to the command that your project is here, in this new dir. DepHell tries to scan this dir and can't find any version. However, you don't see the error because dephell think that "serial" (you passed it as positional argument) is the new version and works with it. This is incorrect. Instead, we make a simple project, write `__version__` inside and after all checks that dephell found this version and bumped last digit (because scheme is `semver` by default and rule is `fix` because we passed it as positional argument).

**Comment by nandahkrishna** [comment]
Ah I guess I misunderstood, thank you for explaining how the test works!

File: dephell/config/scheme.py [hunk 3]
```diff
@@ -60,6 +60,7 @@
 
     # other
     'owner': dict(type='string', required=False),
+    'tag': dict(required=False),
```
Comment (orsinium) [inline dephell/config/scheme.py]: Please, add `type='boolean', ` here

File: tests/test_commands/test_project_bump.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+from dephell.config import Config
+
+
+def test_bump_command(temp_path: Path):
+    (temp_path / 'project').mkdir()
+    init_path = (temp_path / 'project' / '__init__.py')
+    init_path.write_text('__version__ = "1.2.3"')
+    config = Config()
+    config.attach({
+        'project': str(temp_path / 'project'),
```
Comment (orsinium) [inline tests/test_commands/test_project_bump.py]: I guess It should work just with `str(temp_path)`.

**Comment by orsinium** [comment]
A few little details, and I will merge it. You've done big work :+1:

**Comment by nandahkrishna** [comment]
Thanks!

**Comment by orsinium** [comment]
Thank you :heart:

**Pull request merged**